pkg/virt-controller/watch/migration.go

```diff
@@ -535,6 +535,19 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 				migrationCopy.Status.Phase = virtv1.MigrationRunning
 			}
 		case virtv1.MigrationRunning:
+			_, exists := pod.Annotations[virtv1.MigrationTargetStartTimestamp]
+			if !exists && vmi.Status.MigrationState.TargetNodeDomainStartTimestamp != nil {
+				key := patch.EscapeJSONPointer(virtv1.MigrationTargetStartTimestamp)
+				patchOps := fmt.Sprintf(`[{ "op": "add", "path": "/metadata/annotations/%s", "value": "%s" }]`,
+					key,
+					vmi.Status.MigrationState.TargetNodeDomainStartTimestamp.String())
+
+				_, err := c.clientset.CoreV1().Pods(pod.Namespace).Patch(context.Background(), pod.Name, types.JSONPatchType, []byte(patchOps), v1.PatchOptions{})
```

Comment:
Is this the correct place ? It feels like migration has just started, not sure if the target VM is really ready to receive traffic at pre-copy and post-copy last migration.

Reply:
This should be the right spot. What we're checking here is that the migration is in progress, and that the qemu guest has started on the target pod `vmi.Status.MigrationState.TargetNodeDomainStartTime`. If both of these conditions are true, then we patch the annotation on the target pod.